### PR TITLE
Migrate to lodash-es

### PR DESCRIPTION
**Pull request by danReynolds**
As called out in #28 , a number of consumers have reached out for us to switch to lodash-es instead of lodash so that we can reduce bundle size. I've gone and replaced lodash in this PR, if someone wants to try it out and let me know if it resolves their bundle size issue let me know and we can move forward with this. Thanks!

### CODE REVIEW

**Comment by sicksid** [comment]
awesome!

**Comment by danReynolds** [comment]
> awesome!

🥳 Did this reduce your bundle size?

File: tests/EntityTypeMap.test.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import _ from "lodash";
```
Comment (FoodProduct) [inline tests/EntityTypeMap.test.ts:1]: I'm surprised a linter is not catching unused imports...

File: tests/ApolloExtensionsClient.test.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
+import { times } from "lodash-es";
```
Comment (FoodProduct) [inline tests/ApolloExtensionsClient.test.ts:1]: Is there a reason sometimes we rename to `_{method}` and sometimes we don't? Should we standardize on a convention?

**Review by FoodProduct — APPROVED** [review]
Thanks for taking this on! I'm interested in hearing how/if this works out for folks.

File: tests/ApolloExtensionsClient.test.ts (hunk 2 above)
Comment (danReynolds) [reply tests/ApolloExtensionsClient.test.ts:1]: I only prefixed it because of a scope conflict for filter I believe

**Pull request merged**